common/src/main/java/org/opensearch/sdk/BulkDataObjectRequest.java

```diff
@@ -0,0 +1,157 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+package org.opensearch.sdk;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import org.opensearch.action.support.WriteRequest.RefreshPolicy;
+import org.opensearch.common.Nullable;
+import org.opensearch.core.common.Strings;
+
+public class BulkDataObjectRequest {
+
+    private final List<DataObjectRequest> requests = new ArrayList<>();
+    private final Set<String> indices = new HashSet<>();
+    private RefreshPolicy refreshPolicy = RefreshPolicy.NONE;
+    private String globalIndex;
+    private String globalTenantId;
```

Comment:
Add a comment: why it's global?

Reply:
This allows specifying it once when creating a bulk request without having to specify it individually for every sub-request.   Including this global value means we can validate the existence of a tenantid before executing, like we do for all the other requests.   We can remove this and then just iterate all the requests and confirm that they have non-null tenant IDs.

Reply:
Removed this and updated code to check all the individual requests.